extras/ping/ping.go

```diff
@@ -572,15 +585,47 @@ func (p *Pinger) processPacket(recv *packet) error {
 	return nil
 }
 
+func (p *Pinger) parseEchoReplyFromICMP(data []byte, from *net.IP) *Packet {
+	replyPacket, err := icmp.ParseMessage(1, data[:])
+	if err != nil {
+		log.Printf("error: %v\n", err)
+		return &Packet{}
+	}
+	if _, ok := replyPacket.Body.(*icmp.Echo); !ok {
+		log.Printf("error: invalid reply type: %v\n", replyPacket)
+		return &Packet{}
+	}
+
+	icmp := layers.ICMPv4{}
+	udp := layers.UDP{}
+	payload := gopacket.Payload{}
+	decoded := []gopacket.LayerType{}
+	parser := gopacket.NewDecodingLayerParser(layers.LayerTypeICMPv4, &icmp, &udp, &payload)
+
+	if err := parser.DecodeLayers(data, &decoded); err != nil {
+		log.Printf("error: %v\n", err)
+		return &Packet{}
+	}
```

Comment:
I admit I am a bit lost in what you are actually doing here. Would you mind adding some comments to the code explaining why you're parsing an ICMP packet and then apparently parsing it again a second time?

Reply:
It would also be useful to know, more in general, _why_ we need a different parsing strategy with wireguard and we should perhaps consider whether we can adopt the same strategy also when using openvpn.

Reply:
Also, I am kinda lost in this thinking about why we're performing wireguard related changes in this library which does not seem to otherwise deal with wireguard. Is this because we're reusing openvpn code for wireguard?